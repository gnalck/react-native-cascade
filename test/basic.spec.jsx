import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import BasicComponent from './BasicComponent';

it('matches the snapshot', () => {
  const tree = renderer
    .create(<BasicComponent />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('creates a style node', () => {
  const com = shallow(<BasicComponent />);
  const hasStylesNode = com.find({className: 'text-large container'});
  expect(hasStylesNode.prop('style')).toMatchObject({color: 'red', fontSize: 24});
});

it("doesn't create style tags for nodes with unmatched selectors", () => {
  const com = shallow(<BasicComponent />);
  const hasStylesNode = com.find({className: 'text-medium'});
  expect(hasStylesNode.prop('style')).toBeUndefined();
});


// TODO: Test to make sure the runtime isn't included when its not needed
