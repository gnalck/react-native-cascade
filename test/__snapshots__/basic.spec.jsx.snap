// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    className="text-large container"
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 24,
      }
    }
  >
    Hello
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    className="text-medium"
    ellipsizeMode="tail"
  >
    World
  </Text>
</View>
`;
